src/basic/time-util.c

```diff
@@ -849,11 +849,11 @@ static int parse_timestamp_impl(const char *t, usec_t *usec, bool with_tz) {
         }
 
 from_tm:
-        x = mktime_or_timegm(&tm, utc);
-        if (x < 0)
+        if (weekday >= 0 && tm.tm_wday != weekday)
                 return -EINVAL;
 
-        if (weekday >= 0 && tm.tm_wday != weekday)
+        x = mktime_or_timegm(&tm, utc);
```

Comment:
I think it was put the other order purposefully: so that the `tm` time was normalized first, before that extra check.

Reply:
mhmm you can check the test case for x=295138314630994. Before this call the tm struct looks completly good, (as returned by strftime). After the call to mktime_or_timegm the struct is filled with wrong data, even the seconds and hours are wrong now ...

Reply:
yes, mktime() normalizes the input and fills in wday, and that's a good thing, and what we want to check for... Your change reverses the whole logic there

Reply:
Okay, if i remove my change about that and use x=295138314630994 instead of line 151 & 252 in test-time.c, it clearly fails, stepping through the function shows that mktime_or_timegm is changing the contents of the struct, to values that are not represented by x after the function call...

Reply:
Okay, debugged a bit more, the problem is that in the normalization the weekday gets increased decreased by one, because the time jump over the 00:00 border due to a CET -> CEST timezone change, which means the weekday will be invalid after that.

Not too sure how we can solve that, other than checking the weekday before the normalization...